### DISCUSSION TITLE
Wiki-Writing Agent

### DISCUSSION THREAD

**Discussion by BradKML** [SmartManoj/Kevin#135] [Ideas]
This idea is borrowed from PKM tools with MarkDown support, like LogSeq and Siyuan. Sometimes when an agent add more content to the codebase, it is better to write documentation to go along with it. How can it be done such that the agent can hop between CodeAct and writing? Or is there a need for dedicated agent along side a CodeAct agent?

**Comment by SmartManoj** [comment]
CodeAct agents itself write documentation if prompted.

**Reply by BradKML** [reply]
Can we make it such that if we tell them, there are 3 tasks, and after each task is finished please write documentation? I can't seem to get it to work like that for Qwen at least.
Also to be even more specific, the ideal would be to include PKM or list of personal notes to the code such that it can learn from experience, and replay those experience when tackling new issues, for both code and writing purposes.

**Reply by SmartManoj** [reply]
Implemented [custom_instructions](https://github.com/SmartManoj/Kevin/commit/2ae6a8ee368c9d9fa51d57d5120df0c6f50e5db3).

[Demo](https://www.all-hands.dev/share?share_id=7c43ea7c9715ed47dae3f3f624f6e715fc49270e3e0c1c7a6f21664a8969dc4e) with `custom_instructions = 'After each task is finished, please write documentation for the task in doc.md'`

**Comment by BradKML** [comment]
@SmartManoj Now imagine
1. the Agent just read the entire codebase and start writing documentation for each file/folder, and then turn that into a more Wiki-like structure
2. an Agent that specialized in writing blog articles on thought process and dissecting problems, rather than just documenting code for the sake of code usage, and that the blog can self-modify to be accurate, plus throw in a few tag features similar to LogSeq or Siyuan or whatever PKMs are out there

Some example PKMs (as note taking and long term memory proxies): 
- Top FOSS Obsidian alternative https://github.com/logseq/logseq
- Chinese PKM https://github.com/siyuan-note/siyuan
- PKM for VSCode https://github.com/foambubble/foam https://github.com/dendronhq/dendron
- https://github.com/reorproject/reor

**Reply by BradKML** [reply]
Definitely hop onto SiYuan and get nb-assistant patched, cus there is high potential as well for documents complementing code https://github.com/jaar23/nb-assistant/issues/6
